src/irmin-pack/unix/stats_intf.ml

```diff
@@ -82,6 +84,109 @@ module File_manager = struct
   [@@deriving irmin]
 end
 
+module Latest_gc = struct
+  type rusage = {
```

Comment:
In my mind there is a difference between stats that are specific to the code (like appended hashes) and the stats that can be recorded from outside (like rusage). Is there a way that the second type of stats could live in the bench/ package and register callbacks that update them? my concern is that the benchmark code gets too much intertwined with the "main" code, it will be harder to maintain both.  

Having a callback set to be the fun.id, would also make it faster when we are not benchmarking the code ? (even though it is probably fast enough to always record some data)

Lastly, this would avoid some code duplication between this files and the one in bench, right?

I'm not fixed on this, its just to have a discussion.